fotoapparat/src/main/java/io/fotoapparat/routine/preview/OneShotPreviewRoutine.kt

```diff
@@ -0,0 +1,15 @@
+package io.fotoapparat.routine.preview
+
+import android.hardware.Camera
+import io.fotoapparat.hardware.Device
+import kotlinx.coroutines.experimental.runBlocking
+
+/**
+ * Sets one shot preview callback - to get single frame from preview stream
+ */
+internal fun Device.setOneshotPreviewCallback(callback: Camera.PreviewCallback) = runBlocking {
+    val camera = awaitSelectedCamera()
+    camera.setOneShotPreviewCallback(callback)
```

Comment:
Calling this will override the `FrameProcessor` if have been set. Therefore it's better that we use the existing interface in the `Configuration` with two sub types: a `ContinuousFrameProcessor` (existing one) and a `SingleContinuousFrameProcessor` (setting this callback)